### PR TITLE
Pvc 546 acceptance tests handle act1 earning events

### PR DESCRIPTION
**Pull request by samuel-dev-dba**

### CODE REVIEW

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/BaseEarningEventMatcher.cs [hunk 1]
```diff
@@ -255,5 +220,80 @@ private static bool MatchEarningPeriods(ReadOnlyCollection<EarningPeriod> actual
...
+                {
+                    Amount = e.Values[tt],
+                    Period = e.DeliveryCalendarPeriod,
+                    PriceEpisodeIdentifier = FindPriceEpisodeIdentifier(e.Values[tt], e, fm36Learner, tt)
+                }).ToList().AsReadOnly()
+            }).ToList().AsReadOnly();
+            return earningsEvent;
+        }
+
+        protected ApprenticeshipContractTypeEarningsEvent MapToApprenticeshipContractTypeEarningsEventWithIncentives(
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/BaseEarningEventMatcher.cs]: This is always called after an object is created. No need to make it protected and called from derived classes.

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/BaseEarningEventMatcher.cs [hunk 2]
```diff
@@ -255,5 +220,80 @@ private static bool MatchEarningPeriods(ReadOnlyCollection<EarningPeriod> actual
 
             return true;
         }
+
+        protected ApprenticeshipContractTypeEarningsEvent MapToApprenticeshipContractTypeEarningsEventWithoutIncentives(
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/BaseEarningEventMatcher.cs]: This is always called after an object is created. No need to make it protected and called from derived classes.

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ApprenticeshipContractType1EarningEventMatcher.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+using SFA.DAS.Payments.Model.Core.Entities;
+using SFA.DAS.Payments.Model.Core.Incentives;
+using SFA.DAS.Payments.Model.Core.OnProgramme;
+using Earning = SFA.DAS.Payments.AcceptanceTests.EndToEnd.Data.Earning;
+
+namespace SFA.DAS.Payments.AcceptanceTests.EndToEnd.EventMatchers
+{
+    public class ApprenticeshipContractType1EarningEventMatcher: BaseEarningEventMatcher
+    {
+        private readonly IList<Earning> earningSpecs;
```
[the remark this answers is not in the document]
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ApprenticeshipContractType1EarningEventMatcher.cs]: There are only 3 methods in the derived classes which are used in the based mather to validate events. We need to add payments to be validated so you need an object, but you cannot create an instance of the base event ApprenticeshipContractTypeEarningsEvent hence need for the derived classes. Happy to change it if there is a better way.

[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ApprenticeshipContractType1EarningEventMatcher.cs]: I agree - lets discuss. We talked about this with Tony yesterday afternoon

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/EarningEventMatcher.cs [hunk 4]
```diff
@@ -90,24 +95,23 @@ protected override IList<EarningEvent> GetExpectedEvents()
 
                     if (aimSpec.AimReference == "ZPROG001" && onProgEarnings.Any())
                     {
-                        var onProgEarning = new ApprenticeshipContractType2EarningEvent
+                        var onProgEarning  = CreateContractTypeEarningsEventEarningEvent();
+                        onProgEarning.CollectionPeriod = collectionPeriod;
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/EarningEventMatcher.cs]: the collection period and ukprn could be moved into the CreateContractTypeEarningsEventEarningEvent(); method as its common between both usages

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/EarningEventMatcher.cs [hunk 5]
```diff
@@ -90,24 +95,23 @@ protected override IList<EarningEvent> GetExpectedEvents()
...
-                            JobId = testSession.JobId,
-                            Learner = learner,
-                            LearningAim = learningAim
-                        };
+                                Amount = e.Values[tt],
+                                Period = (byte) e.DeliveryCalendarPeriod,
+                                PriceEpisodeIdentifier = FindPriceEpisodeIdentifier(e.Values[tt], e, fm36Learner, tt)
+                            }).ToList().AsReadOnly()
+                        }).ToList().AsReadOnly();
+                        onProgEarning.JobId = testSession.JobId;
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/EarningEventMatcher.cs]: These three variables could be moved to CreateContractTypeEarningsEventEarningEvent() as they are common to both usages

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/EarningEventMatcher.cs (hunk 4 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/EarningEventMatcher.cs]: Done

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/EarningEventMatcher.cs (hunk 5 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/EarningEventMatcher.cs]: Done

**Pull request merged**